ts/a11y/speech/GeneratorPool.ts

```diff
@@ -0,0 +1,353 @@
+ */
+
+export class GeneratorPool<N, T, D> {
+
+  private _element: Element;
+
+  set element(element: Element) {
+    this._element = element;
+    // We always force a rebuild of the semantic tree, in case the element was
+    // re-rendered. Otherwise we might have incorrect maction links etc.
+    const rebuilt = this.speechGenerator.computeRebuilt(element, true);
+    this.brailleGenerator.setRebuilt(rebuilt);
+    this.summaryGenerator.setRebuilt(rebuilt);
+  }
+
+  get element() {
+    return this._element;
+  }
+
+  /**
+   * The adaptor to work with typeset nodes.
+   */
+  public adaptor: DOMAdaptor<N, T, D> = null;
+
+  /**
+   * The speech generator for a math item.
+   */
+  public speechGenerator = Sre.getSpeechGenerator('Tree');
+
+  /**
+   * The braille generator for a math item.
+   */
+  public brailleGenerator = Sre.getSpeechGenerator('Tree');
+
+  /**
+   * The summary generator for a math item.
+   */
+  public summaryGenerator = Sre.getSpeechGenerator('Summary');
+
+  /**
+   *  The current speech setting for Sre
+   */
+  private currentLocale = 'none';
+  private currentBraille = 'none';
+  private _options: OptionList = {};
+
+  /**
+   * Option setter that takes care of setting up SRE and assembling the options
+   * for the speech generators.
+   * 
+   * @param {OptionList} options The option list.
+   */
+  public set options(options: OptionList) {
+    this._options = options;
+    Sre.setupEngine(options.sre);
+    this.speechGenerator.setOptions(Object.assign(
+      {}, options?.sre || {}, {
+        modality: 'speech',
+        markup: 'ssml',
+        automark: true
+      }));
+    this.summaryGenerator.setOptions(Object.assign(
+      {}, options?.sre || {}, {
+        modality: 'summary',
+        markup: 'ssml',
+        automark: true,
+      }));
+    this.brailleGenerator.setOptions({
+      locale: options?.sre?.braille,
+      domain: 'default',
+      style: 'default',
+      modality: 'braille',
+      markup: 'none',
+    });
+  }
+
+  public get options() {
+    return this._options;
+  }
+
+  private _init = false;
+
+  /**
+   * Init method for speech generation. Runs a retry until locales have been
+   * loaded.
+   *
+   * @param {OptionList} options A list of options.
+   */
+  public init(options: OptionList, adaptor: DOMAdaptor<N, T, D>) {
+    if (this._init) return;
+    this.adaptor = adaptor;
+    this.options = options;
+    this._init = true;
+    if (this._update(options)) {
+      mathjax.retryAfter(Sre.sreReady());
+    }
+  }
+
+  /**
+   * Update method for speech generation options. Runs a retry until locales have been
+   * loaded.
+   *
+   * @param {OptionList} options A list of options.
+   */
+  public update(options: OptionList) {
+    this.options = options;
+    return this._update(options);
+  }
+
+  private _update(options: OptionList) {
+    let update = false;
+    if (options?.sre?.braille !== this.currentBraille) {
+      this.currentBraille = options.sre.braille;
```

Comment:
If `options` or `options.sre` don't exist, then the `if` condition will be true, and you will end up with an error from `options.sre.braille` on line 141.  I'm not sure the question marks are needed in the `if` condition, but if you need to be concerned about that, I would suggest something like

``` js
if (!options || !options.sre) return false;
```

at the top.  Similarly for line 145.